fe/src/main/java/org/apache/doris/qe/ShowExecutor.java

```diff
@@ -1146,14 +1147,69 @@ private void handleShowTablet() throws AnalysisException {
                 }
 
                 OlapTable olapTable = (OlapTable) table;
-
-                for (Partition partition : olapTable.getPartitions()) {
+                long sizeLimit = -1;
+                if (showStmt.hasOffset() && showStmt.hasLimit()) {
+                    sizeLimit = showStmt.getOffset() + showStmt.getLimit();
+                } else if (showStmt.hasLimit()) {
+                    sizeLimit = showStmt.getLimit();
+                }
+                boolean stop = false;
+                Collection<Partition> partitions = new ArrayList<Partition>();
+                List<String> partitionNames = showStmt.getPartitionNames();
+                if (showStmt.hasPartition()) {
+                    for (Partition partition : olapTable.getPartitions()) {
+                        if (partitionNames.contains(partition.getName())) {
+                            partitions.add(partition);
+                        }
+                    }
+                } else {
+                    partitions = olapTable.getPartitions();
+                }
+                List<List<Comparable>> tableInfos =  new ArrayList<List<Comparable>>();
+                String indexName = showStmt.getIndexName();
```

Comment:
the user specified index may not exist, so that `olapTable.getIndexIdByName()` will return NULL.
So we should check the index name here and throw exception if index does not exist.

Reply:
ok